### ISSUE TITLE
Error: "rules" contains an invalid JSON; while referencing akamai_cp_code in template_file 

### ISSUE THREAD

**Issue by hanburger97** [akamai/terraform-provider-akamai#159]
Hi there 👋 

### TLDR;
Unable to `apply` tf configuration with a dynamically resolved cp code reference to akamai_cp_code resource inside a template file for JSON rules, despite being able to run successful plan.

### Context

We are trying to configure our akamai property in a way that will not require hard-coded cp code, therefore, we are using a reference to the akamai_cp_code resource inside our rules.json template file and render such during the `apply` phase.

With this we are able to run successful `plan` since nothing is resolved at this stage, however, we are unable to apply them successfully.

### Terraform Version
```
Terraform v0.12.6
+ provider.akamai v0.1.4
+ provider.fastly v0.15.0
+ provider.template v2.1.2
```

### Affected Resource(s)
Please list the resources as a list, for example:
- akamai_property
- akamai_cp_code
- template_file

### Terraform Configuration Files
`property.tf`
```hcl
provider "akamai" {
 edgerc = "~/.edgerc"
 papi_section = "terraform"
}

data "akamai_group" "group" {
 name = "(Removed for privacy)"
}

data "akamai_contract" "contract" {
 name = "(Removed for privacy)"
}

data "template_file" "rules" {
 template = file("${path.module}/rules.json")
 vars = {
   cpcode_id = akamai_cp_code.edgescale.id
 }
}

resource "akamai_cp_code" "edgescale" {
 product  = "prd_Site_Accel"
 contract = data.akamai_contract.contract.id
 group = data.akamai_group.group.id
 name = "edgescale-akamaki-shopifycdn-com"
}

resource "akamai_property" "edgescale" {
 name = "edgescale"
 cp_code = akamai_cp_code.edgescale.id
 contact = [""]
 contract = data.akamai_contract.contract.id
 group = data.akamai_group.group.id
 product = "prd_Site_Accel"
 rule_format = "latest"
 hostnames = {
  "(removed for privacy)" = "(removed for privacy)"
 }
 rules = data.template_file.rules.rendered
 is_secure = false
}

```

rules.json
```json
{
  "accountId": "(removed for privacy)",
  "contractId": "(removed for privacy)",
  "groupId": "(removed for privacy)",
  "propertyId": "(removed for privacy)",
  "propertyVersion": 10,
  "etag": "(removed for privacy)",
  "ruleFormat": "latest",
  "rules": {
    "name": "default",
    "behaviors": [
      {
        "name": "origin",
        "options": {
          ...
        }
      },
      {
        "name": "cpCode",
        "options": {
          "value": {
            "cpCodeLimits": {
              "currentCapacity": 79,
              "limit": 100,
              "limitType": "global"
            },
            "createdDate": 1574351173000,
            "description": "edgescale-akamaki-shopifycdn-com",
            "id": ${cpcode_id},
            "name": "(removed for privacy)",
            "products": [
              "Site_Accel"
            ]
          }
        }
      },
     ...
    ],
    "children": [
          ... 
    ]
  }
}
```

### Debug Output
Debug logs [here](https://gist.github.com/hanburger97/e881f363f7840e449279d9665c3fd405)

### Panic Output
n / a

### Expected Behavior
During the apply phase the provider shall resolve the akamai_cp_code.edgescale resources first as it is dependency for akamai_propert.edgescale resource, once the value computed, the akamai_cp_code.edgescale.id should be computed and resolved, and the rules template_file rendering shall work as expected and replace the cpcode_id variable with the actual value.

### Actual Behavior

```
Error: "rules" contains an invalid JSON: invalid character 'c' looking for beginning of value

  on property.tf line 37, in resource "akamai_property" "edgescale":
  37: resource "akamai_property" "edgescale" {
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Adding a `cpcode_id` variable inside the rules template file resource and set the reference to `akamai_cp_code.edgescale.id` 
2. Update the hard coded cp code ids in the `rules.json` file to `"id": ${cpcode_id}` (See code above)
3. Terraform plan & apply

### Important Factoids
n/a

### References
n/a

**Comment by martinstibbe** [comment]
@hanburger97 Can you please replicate same issue using latest version of provider v0.7.1 
As there have been many changes in provider around properties/rules
I will review similar configuration in the latest version as well

**Comment by hanburger97** [comment]
@martinstibbe yes the same issue is replicated with v0.7.1

**Comment by martinstibbe** [comment]
Sam from Akamai had tried this change to add quotes in json file around variable
"id": "${cpcode_id}" 
in rules.json and I verified it shows up in state file rendered properly

template  \"id\": \"${cpcode_id}\"\n
rendered  \"id\": \"cpc_860373\"\n

**Comment by zstlaw** [comment]
Closing this issue due to lack of response. 

The provider has been revamped in the 1.0.0 release. A more consumable templating capability has been added. Please upgrade to 1.0.0. Please refer https://registry.terraform.io/providers/akamai/akamai/latest/docs/guides/1.0_migration for additional details. 

If you encountered an issue with 1.0.0, for your specific scenario, please submit a new issue.

**Comment by BAu0002** [comment]
I have similar issue to this.
The main.json file have this snip of code:
` {
        "name": "cpCode",
        "options": {
          "value": {
            "cpCodeLimits": {
              "currentCapacity": 77,
              "limit": 100,
              "limitType": "global"
            },
            "id": "${env.cpCodeId}",
            "name": "fake",
            "products": [
              "Fresca"
            ]
          }
        }
      }`
I'm getting the following message 
`"errors": [
│               {
│                       "location": "/rules/behaviors/1/options/value/id",
│                       "schemaLocation": "/definitions/catalog/option_types/cpcode/properties/id",
│                       "detail": "instance type (string) does not match any allowed primitive type (allowed: [\"integer\",\"number\"])",
│                       "foundType": "string",
│                       "allowedTypes": [
│                               "integer",
│                               "number"
│                       ]
│               }
│       ],`

If I remove the double quote around ${env.cpCodeId} in the json file, I get an error saying 
`invalid JSON result: invalid character '$' looking for beginning of value`
